core/src/main/java/org/apache/iceberg/io/SortedPosDeleteWriter.java

```diff
@@ -28,15 +27,17 @@
 import org.apache.iceberg.DeleteFile;
 import org.apache.iceberg.FileFormat;
 import org.apache.iceberg.StructLike;
+import org.apache.iceberg.deletes.PositionDelete;
 import org.apache.iceberg.deletes.PositionDeleteWriter;
 import org.apache.iceberg.encryption.EncryptedOutputFile;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.types.Comparators;
 import org.apache.iceberg.util.CharSequenceSet;
 import org.apache.iceberg.util.CharSequenceWrapper;
 
-class SortedPosDeleteWriter<T> implements Closeable {
+class SortedPosDeleteWriter<T> implements FileWriter<PositionDelete<T>, DeleteWriteResult> {
```

Comment:
@openinx @rdblue, this writer is slightly different from other delete writers we have. Currently, we cannot construct it using `WriterFactory`. I did not add it because it is not possible to build this writer via `Parquet`, `Avro` classes. I'd like to know your thoughts whether we should extend `Parquet`, `Avro`, `WriterFactory` classes to support this one or we consider its purpose fairly narrow so that we will still build `FileAppender` manually and not through `WriterFactory`.

Reply:
Should this wrap an `OrderedPositionDeleteWriter` instead of managing its own appender? Would that help the issue?

Reply:
I think it will. I'll update.